homeassistant/components/zimi/entity.py

```diff
@@ -0,0 +1,66 @@
+"""Base entity for zimi integrations."""
+
+from __future__ import annotations
+
+import logging
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZimiEntity(Entity):
+    """Representation of a Zimi API entity."""
+
+    _attr_should_poll = False
+    _attr_has_entity_name = False
+
+    def __init__(self, entity: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an HA Entity which is a ZimiDevice."""
+
+        self._entity = entity
+        self._attr_unique_id = self._entity.identifier
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._entity.manufacture_info.identifier)},
+            manufacturer=self._entity.manufacture_info.manufacturer,
+            model=self._entity.manufacture_info.model,
+            name=self._entity.manufacture_info.model,
+            hw_version=entity.manufacture_info.hwVersion,
+            sw_version=entity.manufacture_info.firmwareVersion,
+            suggested_area=entity.room,
+            via_device=(DOMAIN, api.mac),
+        )
+        self._attr_name = self._entity.name.strip()
+        self._attr_suggested_area = self._entity.room
+
+    @property
+    def available(self) -> bool:
+        """Return True if Home Assistant is able to read the state and control the underlying device."""
+        return self._entity.is_connected
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to the events."""
+        await super().async_added_to_hass()
+        self._entity.subscribe(self)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cleanup ZimiLight with removal of notification prior to removal."""
+        self._entity.unsubscribe(self)
+        await super().async_will_remove_from_hass()
+
+    def notify(self, _observable: object):
```

Comment:
Why do we use `object` as type annotation? Isn't there a better type to represent `_observable`?

```suggestion
    def notify(self, _observable: object) -> None:
```